source/regression2.Rmd

```diff
@@ -292,21 +291,35 @@ sale price based off of the predictor of home size? Again, answering this is
 tricky and requires knowledge of how you intend to use the prediction.
 
 To visualize the simple linear regression model, we can plot the predicted house
-sale price across all possible house sizes we might encounter superimposed on a scatter
-plot of the original housing price data. There is a plotting function in 
-the `tidyverse`, `geom_smooth`, that
-allows us to add a layer on our plot with the simple
-linear regression predicted line of best fit. By default `geom_smooth` adds some other information
-to the plot that we are not interested in at this point; we provide the argument `se = FALSE` to
-tell `geom_smooth` not to show that information. Figure \@ref(fig:08-lm-predict-all) displays the result.
+sale price across all possible house sizes we might encounter.
+Since our model is linear,
+we only need to compute the predicted value of the min and max points,
+and then connect them with a straight line.
+We superimpose this prediction line on a scatter
+plot of the original housing price data,
+so that we can qualitatively assess if the model seems to fit the data well.
+Figure \@ref(fig:08-lm-predict-all) displays the result.
 
 ```{r 08-lm-predict-all, fig.height = 3.5, fig.width = 4.5, warning = FALSE, fig.pos = "H", out.extra="", message = FALSE, fig.cap = "Scatter plot of sale price versus size with line of best fit for the full Sacramento housing data."}
+sqft_prediction_grid <- tibble(
+    sqft = c(
+        sacramento |> select(sqft) |> min(),
+        sacramento |> select(sqft) |> max()
+    )
+)
+
+sacr_preds <- lm_fit |>
+  predict(sqft_prediction_grid) |>
+  bind_cols(sqft_prediction_grid)
+
 lm_plot_final <- ggplot(sacramento_train, aes(x = sqft, y = price)) +
   geom_point(alpha = 0.4) +
+  geom_line(data = sacr_preds, 
+            mapping = aes(x = sqft, y = .pred), 
+            color = "blue") +
```

Comment:
I believe it is redundant to re-specify any mappings inside `aes` in the geom that are the same as the mapping inside `aes` in the initial `ggplot()` call. The explicit mention of the `mapping = ` and `data = ` parameters here is also redundant and different from how they are not written out inside `ggplot()`. However, in both these cases, this is what is done in the rest of the book as well it seems, so I left it as is for consistency.